src/mongocxx/client.cpp

```diff
@@ -93,7 +95,7 @@ cursor client::list_databases() const {
     auto result = libmongoc::client_find_databases(_impl->client_t, &error);
 
     if (!result) {
-        throw exception::operation(std::make_tuple(error.message, error.code));
+        throw exception::operation{exception::error_code_from_bson_error_t(error), error.message};
```

Comment:
I did it to try to match the constructor signature of `std::system_error`:
`system_error( std::error_code ec, const std::string& what_arg );`
but if you think it's better to keep it the other way, I can switch it back.